torch/_inductor/codegen/triton.py

```diff
@@ -1171,6 +1172,49 @@ def codegen_body(self):
         self.stores.clear()
         self.suffix.clear()
 
+    def codegen_kernel_benchmark(self):
+        result = IndentedBuffer()
+
+        result.writelines(["", "", 'if __name__ == "__main__":'])
+        grid = []
+        with result.indent():
+            result.writeline("from torch._C import _cuda_getCurrentRawStream as get_cuda_stream")
+            result.writeline("from torch._dynamo.testing import rand_strided")
+            result.writeline("from torch._inductor.utils import print_performance")
+            result.writeline("import torch")
+            result.writeline("from torch._inductor.triton_ops.autotune import grid")
+            result.writeline("")
+
+            argdefs, call_args, signature = self.args.python_argdefs()
+
+            for arg_name in call_args:
+                buf = V.graph.get_buffer(arg_name)
+                if buf:
+                    result.writeline(
+                        f"{arg_name} = rand_strided({tuple(buf.get_size())}, {tuple(buf.get_stride())}, device='{buf.get_device()}', dtype={buf.get_dtype()})"
```

Comment:
this won't be exactly right for aliased inputs/outputs, but I don't think it has material impact on perf